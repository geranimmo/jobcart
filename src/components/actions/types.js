export const DISPATCH_LOGIN = 'dispatch_login';
export const DISPATCH_PROFILE_DATA = 'dispatch_profile_data';
export const DISPATCH_SHOPPING_CART = 'dispatch_shopping_cart';
export const DISPATCH_PACKAGES_LIST = 'dispatch_packages_list';
export const ADD_TO_CART = 'add_to_cart';
export const REMOVE_FROM_CART = 'remove_from_cart';
export const DISPATCH_TOTAL_COST = 'dispatch_total_cost';
export const DISPATCH_TOTAL_DISCOUNT = 'dispatch_total_discount';
export const DISPATCH_TESTIMONIAL_LIST = 'dispatch_testimonial_list';