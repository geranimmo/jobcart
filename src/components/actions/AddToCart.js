import { ADD_TO_CART } from './types';

export const addToCart = (datas) => {
	return (dispatch) => {
		let DateTime = new Date().getTime();
		if ( datas.date ) {
			DateTime = datas.date;
		}

		const requestedDatas = {
			id: datas.id,
			buy_time: DateTime
		};
		
		dispatch({
			type: ADD_TO_CART,
			payload: requestedDatas
		});
	};
};