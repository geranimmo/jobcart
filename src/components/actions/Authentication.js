import {
	DISPATCH_LOGIN,
	DISPATCH_PROFILE_DATA,
	DISPATCH_SHOPPING_CART
} from './types';
import AccountList from '../../../src/assets/json/accounts__.json';

export const loginFetch = (datas) => {
	return (dispatch) => {
		const matchAccountData = AccountList.filter((obj) => {
			return obj.email_address === datas.username && obj.user_password === datas.password;
		});
		const newData = new Object(matchAccountData[0]);
		const profileData = {
			client_id: newData.client_id,
			first_name: newData.first_name,
			last_name: newData.last_name,
			business_name: newData.first_name,
			phone_number: newData.phone_number,
			client_logo: newData.client_logo,
			client_special: newData.client_special
		};
		
		dispatch({
			type: DISPATCH_LOGIN,
			payload: matchAccountData.length > 0
		});
		dispatch(dispatchProfile(profileData));
		dispatch(dispatchShoppingCart(newData.shopping_cart));
	};
};

export const dispatchProfile = (datas) => {
	return (dispatch) => {
		dispatch({
			type: DISPATCH_PROFILE_DATA,
			payload: datas
		});
	};
};

export const dispatchShoppingCart = (datas) => {
	return (dispatch) => {
		dispatch({
			type: DISPATCH_SHOPPING_CART,
			payload: datas
		});
	};
};