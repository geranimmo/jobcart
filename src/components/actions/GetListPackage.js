import {
	PACKAGES_LIST_FETCH,
	PACKAGES_LIST_FETCH_SUCCESS,
	PACKAGES_LIST_FETCH_FAILED
} from './types';
import ListPackages from '../../assets/json/packages__.json';

export const packageFetch = () => {
	return (dispatch) => {
		dispatch({ type: PACKAGES_LIST_FETCH });

		if (ListPackages) {
			packageFetchSuccess(dispatch, ListPackages);
		} else {
			packageFetchFailed(dispatch);
		}
	};
};

const packageFetchSuccess = (dispatch, datas) => {
	dispatch({
		type: PACKAGES_LIST_FETCH_SUCCESS,
		payload: datas
	});
};

const packageFetchFailed = (dispatch) => {
	dispatch({ type: PACKAGES_LIST_FETCH_FAILED });
};