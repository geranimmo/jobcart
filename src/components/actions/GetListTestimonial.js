import { DISPATCH_TESTIMONIAL_LIST } from './types';
import ListTestimonials from '../../../src/assets/json/testimonials__.json';

export const getListTestimonial = () => {
	return (dispatch) => {
		dispatch({
			type: DISPATCH_TESTIMONIAL_LIST,
			payload: ListTestimonials
		});
	};
};