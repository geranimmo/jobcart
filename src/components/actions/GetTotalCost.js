import { DISPATCH_TOTAL_COST } from './types';

export const getTotalCost = () => {
	return (dispatch, getState) => {
		const { ShoppingCart } = getState();
		let totalCost = 0;
	
		if ( ShoppingCart.length > 0 ) {
			for ( let i = 0; i < ShoppingCart.length; i++ ) {
				totalCost = parseFloat((totalCost + ShoppingCart[i].package_price).toFixed(2));
			}
		}
        
		dispatch({
			type: DISPATCH_TOTAL_COST,
			payload: totalCost
		});
	};
};