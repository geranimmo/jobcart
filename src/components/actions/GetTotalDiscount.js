import { DISPATCH_TOTAL_DISCOUNT } from './types';

export const getTotalDiscount = () => {
	return (dispatch, getState) => {
		const { ShoppingCart, Profile } = getState();
		const ClientSpecial = Profile.client_special;
		let totalDiscount = 0;
	
		for ( let i = 0; i < ClientSpecial.length; i++ ) {
			let currentDiscount;
			const getInCartPromoItems = filterPromoById(ClientSpecial[i].package_id, ShoppingCart);
			
			if ( ClientSpecial[i].types === 'free__items' ) {
				if (getInCartPromoItems.length) {
					let pricePerItem = getInCartPromoItems[0].package_price;
					let getTotalFreeItems = Math.floor(getInCartPromoItems.length / ClientSpecial[i].minimum_items);
					currentDiscount = parseFloat((getTotalFreeItems * pricePerItem).toFixed(2));
					totalDiscount = totalDiscount + currentDiscount;
				}
			} else if (getInCartPromoItems.length >= ClientSpecial[i].minimum_items) {
				let discountPrice = ClientSpecial[i].discount_price;
				currentDiscount = parseFloat((discountPrice * getInCartPromoItems.length).toFixed(2));
				totalDiscount = totalDiscount + currentDiscount;
			}
		}
        
		dispatch({
			type: DISPATCH_TOTAL_DISCOUNT,
			payload: totalDiscount
		});
	};
};

export const filterPromoById = (id, datas) => {
	return datas.filter(obj => obj.package_id === id);
};