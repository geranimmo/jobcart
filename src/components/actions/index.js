export * from './Authentication';
export * from './GetListPackage';
export * from './AddToCart';
export * from './RemoveFromCart';
export * from './GetTotalCost';
export * from './GetTotalDiscount';
