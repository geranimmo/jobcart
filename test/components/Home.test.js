import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { mount, shallow } from 'enzyme';
import expect from "expect";
// import sinon from 'sinon';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ConnectedHome, { Home } from '../../src/components/home';
import * as actions from '../../src/components/actions';
import * as types from '../../src/components/actions/types';
import ListPackages from '../../src/assets/json/packages__.json';

describe('>>> H O M E ---- Test & Snapshot <<<', () => {
	const mockStore = configureMockStore([thunk]);
	let wrapper, store;

	const propHome = {
		ShoppingCart: [
			{ id: 'Classic', buy_time: 1537425742947 },
			{ id: 'Premium', buy_time: 1537425742948 }
		],
		Profile: {
			client_id: 1,
			first_name: "Default",
			last_name: "Client",
			business_name: "Default",
			phone_number: "+612529291002",
			client_logo: "client-default-logo.png",
			client_special: []
		},
		UserLogin: true,
		Packages: ListPackages,
		id: 'home__page__wrapper'
	};

	beforeEach(() => {
		store = mockStore(propHome);
		wrapper = mount(
			<Provider store={store}>
				<Router>
					<ConnectedHome {...propHome}/>
				</Router>
			</Provider>
		);
	});

	it('+++ Render the Home component', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('+++ Contains props that assigned to the Home component', () => {
		expect(wrapper.find(`main`).prop('id'))
			.toEqual(propHome.id);
	});
    
	it('+++ Check ShoppingCart action in Home ', () => {
		const addCartData = {
			id: "Classic",
			buy_time: 1537425742447
		};
		let action;

		store.dispatch(actions.addToCart(addCartData));
		action = store.getActions();
		
		expect(action[0].type)
			.toBe(types.DISPATCH_PACKAGES_LIST);

		expect(action[1].type)
			.toBe(types.ADD_TO_CART);
	});

	it('+++ Capturing Snapshot of Home component +++', () => {
		const renderedValue = renderer.create(
			<Provider store={store}>
				<Router>
					<ConnectedHome {...propHome}/>
				</Router>
			</Provider>
		).toJSON();
		
		expect(renderedValue).toMatchSnapshot();
	});

	it('+++ Simulate Click Add To Cart in Packages +++', () => {
		const addToCart = jest.fn();

		expect(
			wrapper
				.find('.slider__content__wrapper')
				.find('.slick-current')
				.find('button.button__buy__items')
				.length
		).toEqual(1);
		
		wrapper
			.find('.slider__content__wrapper')
			.find('.slick-current')
			.find('button.button__buy__items')
			.simulate('click');

		expect(addToCart).toEqual(expect.any(Function));
	});

	it('+++ Simulate componentWillReceiveProps when initialize getListPackage() +++', () => {
		const history = { push: jest.fn() };
		const getListPackage = jest.fn();
		const wrappers = shallow(
			<Home
				{...propHome}
				history={history}
				getListPackage={getListPackage}
			/>
		);
		const showCartSpy = jest.spyOn(wrappers.instance(), 'createDataSource');

		wrappers.setProps({
			getListPackage: [{
				"package_id": "Standout"
			}]
		});

		wrappers.instance().forceUpdate();
		wrappers.update();

		expect(showCartSpy.mock.calls.length).toBe(1);
	});

	it('+++ Should call handle scroll function when scroll +++', () => {
		const handleHeaderOnScroll = jest.fn();

		wrapper.find('div#content__scroller').simulate('scroll', { deltaY: 50 });
		expect(handleHeaderOnScroll).toEqual(expect.any(Function));
	});
});