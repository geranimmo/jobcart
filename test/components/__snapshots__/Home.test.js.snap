// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> H O M E ---- Test & Snapshot <<< +++ Capturing Snapshot of Home component +++ 1`] = `
<div
  className="main__container"
>
  <header
    className="header__active"
    id="header"
  >
    <nav
      className="nav__left"
    >
      <a
        aria-current={null}
        href="/home"
        onClick={[Function]}
      >
        <div
          className="seek__logo"
        />
      </a>
      <div
        className="client__logo__wrapper "
      >
        <img
          alt="Default"
          src="client-default-logo.png"
        />
      </div>
    </nav>
    <nav
      className="nav__right "
    >
      <div
        className="menu__btn__icon "
        id="openCartView"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24 3l-.743 2h-1.929l-3.474 12h-13.239l-4.615-11h16.812l-.564 2h-13.24l2.937 7h10.428l3.432-12h4.195zm-15.5 15c-.828 0-1.5.672-1.5 1.5 0 .829.672 1.5 1.5 1.5s1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5zm6.9-7-1.9 7c-.828 0-1.5.671-1.5 1.5s.672 1.5 1.5 1.5 1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5z"
          />
        </svg>
        <div
          className="cart__notif "
        >
          2
        </div>
      </div>
      <div
        className="menu__btn__icon hidden"
        id="closeCartView"
        onClick={[Function]}
      >
        <svg
          className="icon__close"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M23.954 21.03l-9.184-9.095 9.092-9.174-2.832-2.807-9.09 9.179-9.176-9.088-2.81 2.81 9.186 9.105-9.095 9.184 2.81 2.81 9.112-9.192 9.18 9.1z"
          />
        </svg>
      </div>
    </nav>
  </header>
  <main
    className="home__component"
    id="home__page__wrapper"
  >
    <div
      className="content__wrapper"
      id="content__scroller"
      onScroll={[Function]}
    >
      <div
        className="slider__container "
      >
        <div
          className="slider__nav__wrapper"
        >
          <div
            className="slick-slider slick-initialized"
          >
            
            <div
              className="slick-list"
            >
              <div
                className="slick-track"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "WebkitTransform": "translate3d(0px, 0px, 0px)",
                    "WebkitTransition": "",
                    "msTransform": "translateX(0px)",
                    "opacity": 1,
                    "transform": "translate3d(0px, 0px, 0px)",
                    "transition": "",
                  }
                }
              >
                <div
                  aria-hidden={false}
                  className="slick-slide slick-active slick-current"
                  data-index={0}
                  onClick={[Function]}
                  style={
                    Object {
                      "outline": "none",
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="slider__card__nav "
                      id="navClassic"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <h1
                        className="no__margin"
                      >
                        Classic
                      </h1>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={false}
                  className="slick-slide slick-active"
                  data-index={1}
                  onClick={[Function]}
                  style={
                    Object {
                      "outline": "none",
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="slider__card__nav nav__recommended"
                      id="navStandout"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <h1
                        className="no__margin"
                      >
                        Standout
                      </h1>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={false}
                  className="slick-slide slick-active"
                  data-index={2}
                  onClick={[Function]}
                  style={
                    Object {
                      "outline": "none",
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="slider__card__nav "
                      id="navPremium"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <h1
                        className="no__margin"
                      >
                        Premium
                      </h1>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            
            
          </div>
        </div>
        <div
          className="slider__content__wrapper"
        >
          <div
            className="slick-slider slick-initialized"
            dir="ltr"
          >
            <div
              className="slick-list"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={null}
              onMouseMove={null}
              onMouseUp={[Function]}
              onTouchCancel={null}
              onTouchEnd={[Function]}
              onTouchMove={null}
              onTouchStart={[Function]}
              style={Object {}}
            >
              <div
                className="slick-track"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "WebkitTransform": "translate3d(0px, 0px, 0px)",
                    "WebkitTransition": "",
                    "msTransform": "translateX(0px)",
                    "opacity": 1,
                    "transform": "translate3d(0px, 0px, 0px)",
                    "transition": "",
                  }
                }
              >
                <div
                  aria-hidden={true}
                  className="slick-slide slick-cloned"
                  data-index={-1}
                  onClick={[Function]}
                  style={
                    Object {
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="section__carousel"
                      id="Premium"
                      itemProp="itemListElement"
                      itemScope={true}
                      itemType="http://schema.org/Product"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <div
                        className="slider__content"
                      >
                        <div
                          className="slider__card__body"
                        >
                          <img
                            alt="Premium"
                            className="slider__images"
                            src="/assets/images/employeer-image-sample3.jpg"
                          />
                          <h2
                            className="no__margin__top block width__100"
                          >
                            Premium Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry
                          </h2>
                          <span
                            className="label__info hidden"
                          >
                            RECOMMENDED
                          </span>
                          <span
                            className="pricing block width__100"
                          >
                            $
                            394.99
                          </span>
                          <button
                            className="button__buy__items"
                            onClick={[Function]}
                            type="button"
                            value="Buy Ad"
                          >
                            Buy Ad
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={false}
                  className="slick-slide slick-active slick-current"
                  data-index={0}
                  onClick={[Function]}
                  style={
                    Object {
                      "outline": "none",
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="section__carousel"
                      id="Classic"
                      itemProp="itemListElement"
                      itemScope={true}
                      itemType="http://schema.org/Product"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <div
                        className="slider__content"
                      >
                        <div
                          className="slider__card__body"
                        >
                          <img
                            alt="Classic"
                            className="slider__images"
                            src="/assets/images/employeer-image-sample.jpg"
                          />
                          <h2
                            className="no__margin__top block width__100"
                          >
                            Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting
                          </h2>
                          <span
                            className="label__info hidden"
                          >
                            RECOMMENDED
                          </span>
                          <span
                            className="pricing block width__100"
                          >
                            $
                            269.99
                          </span>
                          <button
                            className="button__buy__items"
                            onClick={[Function]}
                            type="button"
                            value="Buy Ad"
                          >
                            Buy Ad
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={true}
                  className="slick-slide"
                  data-index={1}
                  onClick={[Function]}
                  style={
                    Object {
                      "outline": "none",
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="section__carousel"
                      id="Standout"
                      itemProp="itemListElement"
                      itemScope={true}
                      itemType="http://schema.org/Product"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <div
                        className="slider__content"
                      >
                        <div
                          className="slider__card__body"
                        >
                          <img
                            alt="Standout"
                            className="slider__images"
                            src="/assets/images/employeer-image-sample2.jpg"
                          />
                          <h2
                            className="no__margin__top block width__100"
                          >
                            Standout Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry of Lorem Ipsum
                          </h2>
                          <span
                            className="label__info "
                          >
                            RECOMMENDED
                          </span>
                          <span
                            className="pricing block width__100"
                          >
                            $
                            322.99
                          </span>
                          <button
                            className="button__buy__items"
                            onClick={[Function]}
                            type="button"
                            value="Buy Ad"
                          >
                            Buy Ad
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={true}
                  className="slick-slide"
                  data-index={2}
                  onClick={[Function]}
                  style={
                    Object {
                      "outline": "none",
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="section__carousel"
                      id="Premium"
                      itemProp="itemListElement"
                      itemScope={true}
                      itemType="http://schema.org/Product"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <div
                        className="slider__content"
                      >
                        <div
                          className="slider__card__body"
                        >
                          <img
                            alt="Premium"
                            className="slider__images"
                            src="/assets/images/employeer-image-sample3.jpg"
                          />
                          <h2
                            className="no__margin__top block width__100"
                          >
                            Premium Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry
                          </h2>
                          <span
                            className="label__info hidden"
                          >
                            RECOMMENDED
                          </span>
                          <span
                            className="pricing block width__100"
                          >
                            $
                            394.99
                          </span>
                          <button
                            className="button__buy__items"
                            onClick={[Function]}
                            type="button"
                            value="Buy Ad"
                          >
                            Buy Ad
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={true}
                  className="slick-slide slick-cloned"
                  data-index={3}
                  onClick={[Function]}
                  style={
                    Object {
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="section__carousel"
                      id="Classic"
                      itemProp="itemListElement"
                      itemScope={true}
                      itemType="http://schema.org/Product"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <div
                        className="slider__content"
                      >
                        <div
                          className="slider__card__body"
                        >
                          <img
                            alt="Classic"
                            className="slider__images"
                            src="/assets/images/employeer-image-sample.jpg"
                          />
                          <h2
                            className="no__margin__top block width__100"
                          >
                            Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting
                          </h2>
                          <span
                            className="label__info hidden"
                          >
                            RECOMMENDED
                          </span>
                          <span
                            className="pricing block width__100"
                          >
                            $
                            269.99
                          </span>
                          <button
                            className="button__buy__items"
                            onClick={[Function]}
                            type="button"
                            value="Buy Ad"
                          >
                            Buy Ad
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={true}
                  className="slick-slide slick-cloned"
                  data-index={4}
                  onClick={[Function]}
                  style={
                    Object {
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="section__carousel"
                      id="Standout"
                      itemProp="itemListElement"
                      itemScope={true}
                      itemType="http://schema.org/Product"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <div
                        className="slider__content"
                      >
                        <div
                          className="slider__card__body"
                        >
                          <img
                            alt="Standout"
                            className="slider__images"
                            src="/assets/images/employeer-image-sample2.jpg"
                          />
                          <h2
                            className="no__margin__top block width__100"
                          >
                            Standout Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry of Lorem Ipsum
                          </h2>
                          <span
                            className="label__info "
                          >
                            RECOMMENDED
                          </span>
                          <span
                            className="pricing block width__100"
                          >
                            $
                            322.99
                          </span>
                          <button
                            className="button__buy__items"
                            onClick={[Function]}
                            type="button"
                            value="Buy Ad"
                          >
                            Buy Ad
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden={true}
                  className="slick-slide slick-cloned"
                  data-index={5}
                  onClick={[Function]}
                  style={
                    Object {
                      "width": 0,
                    }
                  }
                  tabIndex="-1"
                >
                  <div>
                    <div
                      className="section__carousel"
                      id="Premium"
                      itemProp="itemListElement"
                      itemScope={true}
                      itemType="http://schema.org/Product"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                    >
                      <div
                        className="slider__content"
                      >
                        <div
                          className="slider__card__body"
                        >
                          <img
                            alt="Premium"
                            className="slider__images"
                            src="/assets/images/employeer-image-sample3.jpg"
                          />
                          <h2
                            className="no__margin__top block width__100"
                          >
                            Premium Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry
                          </h2>
                          <span
                            className="label__info hidden"
                          >
                            RECOMMENDED
                          </span>
                          <span
                            className="pricing block width__100"
                          >
                            $
                            394.99
                          </span>
                          <button
                            className="button__buy__items"
                            onClick={[Function]}
                            type="button"
                            value="Buy Ad"
                          >
                            Buy Ad
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      ;
    </div>
  </main>
</div>
`;
